compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -348,6 +362,47 @@ public static ServiceDeclarationNode getServiceDeclarationNode(Node node, Semant
         return serviceDeclarationNode;
     }
 
+    public static ClassDefinitionNode getServiceClassDefinitionNode(SyntaxNodeAnalysisContext ctx) {
+        if (ctx.node().kind() != CLASS_DEFINITION) {
+            return null;
+        }
+        ClassDefinitionNode classDefinitionNode = (ClassDefinitionNode) ctx.node();
+        Optional<Symbol> serviceContractType = ctx.semanticModel().types()
```

Comment:
I see that the http module has both `http:Service` and `http:ServiceContract` as types. Since `Service` objects are also contracts, I was referring to the type of this contract. Since this could be confusing, I'll refactor it for better clarity.

Reply:
fixed [2aa962f](https://github.com/ballerina-platform/module-ballerina-http/pull/2202/commits/2aa962f18f1fcf259c65a9deae84606a1b2b4dad)